src/regions/regions.js

```diff
@@ -153,4 +155,35 @@ export default class Regions {
 
         this.regions_info.history.redoHistory();
     }
+
+    /**
+     * Handles Click on ROI link button
+     *
+     * @memberof Regions
+     */
+    toggleShowRoiLink() {
+        this.show_roi_link = !this.show_roi_link;
+        if (this.show_roi_link) {
+            setTimeout(() => {document.getElementById('roi_link_input').select()}, 0);
+        }
+    }
+
+    /**
+     * Used by template to provide link to currently selected ROIs
+     *
+     * @param {Array} shapes from regions_info.selected_shapes 'roi:shape' IDs
+     * @memberof Regions
+     */
+    getRoisLink(shapes) {
+        if (!shapes || shapes.length === 0) {
+            return "No shapes selected";
+        }
+        // Only take first ROI:shape
+        let shapeId = shapes[0].split(':')[1];
+        // If unsaved, will have negative IDs. Return message instead
+        if (shapeId < 0) {
+            return "Can't link to unsaved ROIs";
```

Comment:
Shouldn't be unsaved Shapes since the link only allows to select shapes

Reply:
I guess it could be either. Since we're in the "ROIs" tab and when you "Save" you're saving ROIs. We don't really use "Shapes" terminology anywhere else in the UI I think.